src/init/variables/VarResolver.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable tsdoc/syntax */
+import yargs from 'yargs';
+import { createErrorMessage } from '../..';
+import { AsyncHandler } from '../../util/handlers/AsyncHandler';
+import { modulePathPlaceholder } from '../../util/PathUtil';
+import type { VarComputer } from './VarComputer';
+
+const defaultConfig = `${modulePathPlaceholder}config/default.json`;
+const defaultVarConfig = `${modulePathPlaceholder}config/app-setup/vars.json`;
+
+export type YargsArgOptions = Record<string, yargs.Options>;
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```

Comment:
Let's fix that one.

Reply:
Here, it is common in yargs builder pattern. It computes complex dynamic type, which contains all the fields we defined dnamically. In return type, otherwise we have enumerate all those fields again.

It just says, implicitly assign yargs-inferred type. This will allow us to use those fields when we are consuming.

Reply:
We can still fix the warning; just specifc a return type on the function; worst case it is `Object`.

Reply:
Then we have to cast every field when we are consuming them in `AppRunner`, for they are dynamic. see converation with @joachimvh

Reply:
We just need anything, really:

```
export function setupBaseArgs(yargv: yargs.Argv): any {
```

or whatever is fine.